Requirement already satisfied: numpy in /users/aca22yn/.local/lib/python3.10/site-packages (2.1.2)
Requirement already satisfied: torch in /users/aca22yn/.local/lib/python3.10/site-packages (2.5.0)
Requirement already satisfied: transformers in /users/aca22yn/.local/lib/python3.10/site-packages (4.45.2)
Requirement already satisfied: datasets in /users/aca22yn/.local/lib/python3.10/site-packages (3.0.1)
Requirement already satisfied: accelerate in /users/aca22yn/.local/lib/python3.10/site-packages (1.0.1)
Requirement already satisfied: nvidia-cublas-cu12==12.4.5.8 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (12.4.5.8)
Requirement already satisfied: nvidia-cufft-cu12==11.2.1.3 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (11.2.1.3)
Requirement already satisfied: filelock in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from torch) (3.8.0)
Requirement already satisfied: jinja2 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from torch) (3.1.2)
Requirement already satisfied: nvidia-nvtx-cu12==12.4.127 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (12.4.127)
Requirement already satisfied: nvidia-cusolver-cu12==11.6.1.9 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (11.6.1.9)
Requirement already satisfied: fsspec in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (2024.6.1)
Requirement already satisfied: triton==3.1.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (3.1.0)
Requirement already satisfied: nvidia-curand-cu12==10.3.5.147 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (10.3.5.147)
Requirement already satisfied: sympy==1.13.1 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (1.13.1)
Requirement already satisfied: nvidia-nvjitlink-cu12==12.4.127 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (12.4.127)
Requirement already satisfied: nvidia-cuda-cupti-cu12==12.4.127 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (12.4.127)
Requirement already satisfied: nvidia-nccl-cu12==2.21.5 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (2.21.5)
Requirement already satisfied: nvidia-cudnn-cu12==9.1.0.70 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (9.1.0.70)
Requirement already satisfied: nvidia-cuda-runtime-cu12==12.4.127 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (12.4.127)
Requirement already satisfied: networkx in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (3.4.1)
Requirement already satisfied: nvidia-cusparse-cu12==12.3.1.170 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (12.3.1.170)
Requirement already satisfied: typing-extensions>=4.8.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (4.12.2)
Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.4.127 in /users/aca22yn/.local/lib/python3.10/site-packages (from torch) (12.4.127)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from sympy==1.13.1->torch) (1.3.0)
Requirement already satisfied: tqdm>=4.27 in /users/aca22yn/.local/lib/python3.10/site-packages (from transformers) (4.66.5)
Requirement already satisfied: requests in /users/aca22yn/.local/lib/python3.10/site-packages (from transformers) (2.32.3)
Requirement already satisfied: huggingface-hub<1.0,>=0.23.2 in /users/aca22yn/.local/lib/python3.10/site-packages (from transformers) (0.26.0)
Requirement already satisfied: tokenizers<0.21,>=0.20 in /users/aca22yn/.local/lib/python3.10/site-packages (from transformers) (0.20.1)
Requirement already satisfied: pyyaml>=5.1 in /users/aca22yn/.local/lib/python3.10/site-packages (from transformers) (6.0.2)
Requirement already satisfied: safetensors>=0.4.1 in /users/aca22yn/.local/lib/python3.10/site-packages (from transformers) (0.4.5)
Requirement already satisfied: regex!=2019.12.17 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from transformers) (2022.10.31)
Requirement already satisfied: packaging>=20.0 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from transformers) (21.3)
Requirement already satisfied: pyarrow>=15.0.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from datasets) (17.0.0)
Requirement already satisfied: aiohttp in /users/aca22yn/.local/lib/python3.10/site-packages (from datasets) (3.10.10)
Requirement already satisfied: pandas in /users/aca22yn/.local/lib/python3.10/site-packages (from datasets) (2.2.3)
Requirement already satisfied: xxhash in /users/aca22yn/.local/lib/python3.10/site-packages (from datasets) (3.5.0)
Requirement already satisfied: dill<0.3.9,>=0.3.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from datasets) (0.3.8)
Requirement already satisfied: multiprocess in /users/aca22yn/.local/lib/python3.10/site-packages (from datasets) (0.70.16)
Requirement already satisfied: psutil in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from accelerate) (5.9.4)
Requirement already satisfied: yarl<2.0,>=1.12.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from aiohttp->datasets) (1.15.5)
Requirement already satisfied: frozenlist>=1.1.1 in /users/aca22yn/.local/lib/python3.10/site-packages (from aiohttp->datasets) (1.4.1)
Requirement already satisfied: attrs>=17.3.0 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from aiohttp->datasets) (22.1.0)
Requirement already satisfied: async-timeout<5.0,>=4.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from aiohttp->datasets) (4.0.3)
Requirement already satisfied: aiosignal>=1.1.2 in /users/aca22yn/.local/lib/python3.10/site-packages (from aiohttp->datasets) (1.3.1)
Requirement already satisfied: multidict<7.0,>=4.5 in /users/aca22yn/.local/lib/python3.10/site-packages (from aiohttp->datasets) (6.1.0)
Requirement already satisfied: aiohappyeyeballs>=2.3.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from aiohttp->datasets) (2.4.3)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from packaging>=20.0->transformers) (3.0.9)
Requirement already satisfied: charset-normalizer<4,>=2 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from requests->transformers) (2.1.1)
Requirement already satisfied: certifi>=2017.4.17 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from requests->transformers) (2022.9.24)
Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from requests->transformers) (1.26.12)
Requirement already satisfied: idna<4,>=2.5 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from requests->transformers) (3.4)
Requirement already satisfied: MarkupSafe>=2.0 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from jinja2->torch) (2.1.1)
Requirement already satisfied: python-dateutil>=2.8.2 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from pandas->datasets) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from pandas->datasets) (2022.6)
Requirement already satisfied: tzdata>=2022.7 in /users/aca22yn/.local/lib/python3.10/site-packages (from pandas->datasets) (2024.2)
Requirement already satisfied: six>=1.5 in /opt/apps/testapps/el7/software/staging/Python/3.10.8-GCCcore-12.2.0/lib/python3.10/site-packages (from python-dateutil>=2.8.2->pandas->datasets) (1.16.0)
Requirement already satisfied: propcache>=0.2.0 in /users/aca22yn/.local/lib/python3.10/site-packages (from yarl<2.0,>=1.12.0->aiohttp->datasets) (0.2.0)

[notice] A new release of pip available: 22.3.1 -> 24.2
[notice] To update, run: pip install --upgrade pip
/users/aca22yn/.local/lib/python3.10/site-packages/transformers/utils/hub.py:128: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.
  warnings.warn(
/users/aca22yn/.local/lib/python3.10/site-packages/transformers/models/auto/auto_factory.py:471: FutureWarning: The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers. Please use `token` instead.
  warnings.warn(
torch 2.5.0
transformers 4.45.2
accelerate 1.0.1
# of gpus:  0
loading llm model meta-llama/Llama-3.2-1B-Instruct
use device  cuda:0
pruning starts
loading calibdation data
The repository for c4 contains custom code which must be executed to correctly load the dataset. You can inspect the repository content at https://hf.co/datasets/c4.
You can avoid this prompt in future by passing the argument `trust_remote_code=True`.

Do you wish to run the custom code? [y/N] Traceback (most recent call last):
  File "/mnt/parscratch/users/aca22yn/wanda/main.py", line 111, in <module>
    main()
  File "/mnt/parscratch/users/aca22yn/wanda/main.py", line 70, in main
    prune_wanda(args, model, tokenizer, device, prune_n=prune_n, prune_m=prune_m)
  File "/mnt/parscratch/users/aca22yn/wanda/lib/prune.py", line 132, in prune_wanda
    dataloader, _ = get_loaders("c4",nsamples=args.nsamples,seed=args.seed,seqlen=model.seqlen,tokenizer=tokenizer)
  File "/mnt/parscratch/users/aca22yn/wanda/lib/data.py", line 73, in get_loaders
    return get_c4(nsamples, seed, seqlen, tokenizer)
  File "/mnt/parscratch/users/aca22yn/wanda/lib/data.py", line 43, in get_c4
    traindata = load_dataset('c4', 'en', split='train')
  File "/users/aca22yn/.local/lib/python3.10/site-packages/datasets/load.py", line 2074, in load_dataset
    builder_instance = load_dataset_builder(
  File "/users/aca22yn/.local/lib/python3.10/site-packages/datasets/load.py", line 1795, in load_dataset_builder
    dataset_module = dataset_module_factory(
  File "/users/aca22yn/.local/lib/python3.10/site-packages/datasets/load.py", line 1671, in dataset_module_factory
    raise e1 from None
  File "/users/aca22yn/.local/lib/python3.10/site-packages/datasets/load.py", line 1638, in dataset_module_factory
    ).get_module()
  File "/users/aca22yn/.local/lib/python3.10/site-packages/datasets/load.py", line 1318, in get_module
    trust_remote_code = resolve_trust_remote_code(self.trust_remote_code, self.name)
  File "/users/aca22yn/.local/lib/python3.10/site-packages/datasets/load.py", line 131, in resolve_trust_remote_code
    raise ValueError(
ValueError: The repository for c4 contains custom code which must be executed to correctly load the dataset. You can inspect the repository content at https://hf.co/datasets/c4.
Please pass the argument `trust_remote_code=True` to allow custom code to be run.
